src/app/backend/integration/manager.go

```diff
@@ -0,0 +1,70 @@
+package integration
+
+import (
+	"fmt"
+
+	"github.com/kubernetes/dashboard/src/app/backend/client"
+	"github.com/kubernetes/dashboard/src/app/backend/integration/api"
+	"github.com/kubernetes/dashboard/src/app/backend/integration/metric"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IntegrationManager is responsible for management of all integrated applications.
+type IntegrationManager interface {
+	// List returns list of all supported integrations.
+	List() []api.Integration
+	// GetState returns state of integration based on its' id.
+	GetState(id api.IntegrationID) (*api.IntegrationState, error)
+	// Metric returns metric manager that is responsible for management of metric integrations.
+	Metric() metric.MetricManager
+}
+
+// Implements IntegrationManager interface
+type integrationManager struct {
+	metric metric.MetricManager
+}
+
+// List implements integration manager interface. See IntegrationManager for more information.
+func (self *integrationManager) List() []api.Integration {
+	result := make([]api.Integration, 0)
+
+	// Append all types of integrations
+	result = append(result, self.Metric().List()...)
```

Comment:
I was thinking about that at some point but this list would be empty in the beginning and would have to be filled by some function. In example to create MetricClient integration for heapster we need host and `ClientManager` args so we can't easily initialize this list. 

Additionally slices in go are not thread-safe so keeping static array and operating on it from multiple places might cause some troubles.